src/v2/view-builder/views/symantec/AuthenticatorSymantecView.js

```diff
@@ -0,0 +1,46 @@
+import { _, loc } from 'okta';
+import BaseForm from '../../internals/BaseForm';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+import AuthenticatorVerifyFooter from '../../components/AuthenticatorVerifyFooter';
+
+const Body = BaseForm.extend({
+
+  title () {
+    return this.options.isChallenge
+      ? loc('oie.authenticator.challenge.title', 'login', [this.options.displayName])
+      : loc('oie.authenticator.enroll.title', 'login', [this.options.displayName]);
+  },
+
+  subtitle () {
+    return this.options.isChallenge
+      ? loc('oie.symantecVip.challenge.description', 'login')
+      : loc('oie.symantecVip.enroll.description', 'login');
+  },
+
+  save () {
+    return this.options.isChallenge
+      ? loc('mfa.challenge.verify', 'login')
+      : loc('mfa.enroll', 'login');
+  },
+
+});
+
+export default BaseAuthenticatorView.extend({
+  initialize () {
+    BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
+
+    const currentAuthenticator = this.options.appState.get('currentAuthenticator');
+    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments')?.value || [];
+    const existingAuthenticator = _.find(authenticatorEnrollments, { id: currentAuthenticator.id });
+
+    // If the authenticator is already enrolled, we're in a challenge flow as opposed to enrollment
+    this.options.isChallenge = existingAuthenticator || false;
+    this.options.displayName = currentAuthenticator.displayName;
```

Comment:
Inferring whether we're in a challenge flow or not by checking the API response is **super valuable**, as we're able to remove duplicate view code. For most authenticators, all this logic could be extracted into the `BaseAuthenticatorView`.

This has a few benefits:

1. Assuming we move toward utilizing the `displayName` over a hardcoded name in the `title`, all `title` attributes can utilize the same `i18n` key. (See conversation above)
2. Whether to **Enroll** or **Verify** is implicitly decided by the API response
3. Code cleanup!
   - There are quite a few authenticator enroll/challenge views that have nearly identical code. For example, the OnPrem authenticator ([enroll](https://github.com/okta/okta-signin-widget/blob/master/src/v2/view-builder/views/on-prem/EnrollAuthenticatorOnPremView.js)/[challenge](https://github.com/okta/okta-signin-widget/blob/master/src/v2/view-builder/views/on-prem/ChallengeAuthenticatorOnPremView.js)), security question ([enroll/challenge](https://github.com/okta/okta-signin-widget/tree/master/src/v2/view-builder/views/security-question)), and IdP (condensed already).

If there is another +1 to this, I'll add a note to bring it up during the next design review.

Reply:
I was going to say that we now have 2 different patterns to create enroll/verify views, so we should choose one! And seems the answer is here. So +1 to bring it up in a design review in case we are missing something from having 2 different files, and if we can stick to it (maybe even refactor the other authenticators views at some point). And yes, I think we should show displayName from the response

Reply:
This is cool, but maybe we should move this computation of `isChallenge` into appState instead of doing it deep at the view level ? That way it is consistent for each view.

Reply:
Thanks for the feedback all! I'll capture this feedback for a future design review. In regards to moving it to the `appState`, I think this makes a **lot** of sense!